astropy/units/docgen.py

```diff
@@ -25,53 +24,26 @@ def _get_first_sentence(s: str) -> str:
     return s.replace("\n", " ")
 
 
-def _iter_unit_summary(
-    namespace: Mapping[str, object],
-) -> Generator[tuple[NamedUnit, str, str, str, Literal["Yes", "No"]], None, None]:
-    """
-    Generates the ``(unit, doc, represents, aliases, prefixes)``
-    tuple used to format the unit summary docs in `generate_unit_summary`.
-    """
-    # Get all of the units, and keep track of which ones have SI
-    # prefixes
-    units = []
-    has_prefixes = set()
-    for key, val in namespace.items():
-        # Skip non-unit items
-        if not isinstance(val, UnitBase):
-            continue
-
-        if not isinstance(val, NamedUnit):
-            raise TypeError(f"{key!r} must be defined with 'def_unit()'")
-
-        # Skip aliases
-        if key != val.name:
-            continue
-
-        if isinstance(val, PrefixUnit):
-            # This will return the root unit that is scaled by the prefix
-            # attached to it
-            has_prefixes.add(val._represents.bases[0].name)
-        else:
-            units.append(val)
-
-    # Sort alphabetically, case insensitive
-    units.sort(key=lambda x: x.name.lower())
-
-    for unit in units:
-        doc = _get_first_sentence(unit.__doc__).strip()
+def _summarize_units(
+    units: Iterable[NamedUnit],
+    have_prefixes: set[str],
```

Comment:
I annotated this as `set[str]` instead of `Container[str]` because `str` is also a subtype of `Container[str]`.